fast64_internal/utility.py

```diff
@@ -991,4 +1163,20 @@ def read16bitRGBA(data):
 	b = bitMask(data,  1, 5) / ((2**5) - 1)
 	a = bitMask(data,  0, 1) / ((2**1) - 1)
 
-	return [r,g,b,a]
\ No newline at end of file
+	return [r,g,b,a]
+
+def join_c_args(args: 'list[str]'):
+	return ', '.join(args)
+
+def translate_xyz_to_xzy(translate: mathutils.Vector):
+	xzy_translate = translate.xzy
+	xzy_translate[2] = -xzy_translate[2]
+	return xzy_translate
+
+def rotation_xyz_quat_to_xzy_quat(rotation: mathutils.Quaternion):
+    euler_rot = rotation.to_matrix().inverted().to_euler('ZXY')
+    new_rot = mathutils.Euler([-euler_rot.x, -euler_rot.z, euler_rot.y], 'ZXY')
+    return new_rot.to_quaternion()
```

Comment:
these look very sketchy, I feel like using the blender_to_n64 matrix instead could clean it up, if that's what is meant to be done here

Reply:
do you know how i should use that here? I can't get any other variation of this function to work, I probably spent like 4 hours on this function alone lmao it was killing me

Reply:
i'm messing with variations of this right now and cant find anything that produces the correct output
```python
def rotation_xyz_quat_to_xzy_quat(rotation: mathutils.Quaternion):
    new_rot = (rotation.to_matrix().to_4x4() @ transform_mtx_blender_to_n64())
    return new_rot.to_quaternion()
```

Reply:
maybe
```py
def rotation_xyz_quat_to_xzy_quat(rotation: mathutils.Quaternion):
    new_rot = (transform_mtx_blender_to_n64() @ rotation.to_matrix().to_4x4() @ transform_mtx_blender_to_n64().invert())
    return new_rot.to_quaternion()
```
assuming `rotation` is in "blender coordinates" and `new_rot` in "n64 coordinates"

Reply:
awesome!! yeah that works perfectly, identical output. `def rotate_quat_blender_to_n64`?

Reply:
sure